File: elementary_computer_science/Python/HTP/even_odd_bit.py
def nhi_phan(x):
	s = ''
	while (x != 0):
		s += str(x % 2)
		x = x // 2
	return s	
def dem_bit1(s):
	d = s.count('1')
	return d
t = int(input())
A = []
for _ in range(t):
	n = int(input())
	if (dem_bit1(nhi_phan(n)) % 2 == 0):
		A += ['even']
	else:
		A += ['odd']
print('\n'.join(A))

File: elementary_computer_science/Cpp/HTP/even_odd_bit.cpp
#include <iostream>
using namespace std;

int dem_bit1(int x){
	int d = 0;
	while (x) {
		d += x % 2;
		x = int(x/2);
	}
	return d;
}
int main() {
	int t, n;
	cin >> t;
	while (t--) {
		cin >> n;
		if (dem_bit1(n) % 2 == 0)
			cout << "even" << '\n';
		else
			 cout << "odd" << '\n';
	}
}
